docs/frontend.md

```diff
@@ -1,32 +1,121 @@
+| ------------------------------------ | ------------------------------- |
+| Shared                               | `app/frontend/styles`           |
+| API docs                             | `app/frontend/styles/api_docs`  |
+| Apply for teacher training           | `app/frontend/styles/candidate` |
+| Manage teacher training applications | `app/frontend/styles/provider`  |
+| Support for apply                    | `app/frontend/styles/support`   |
+
+We keep imports ordered alphabetically (using BEM notation means styles are encapsulated, meaning the order in which styles are imported shouldn’t matter).
+
+## JavaScript
+
+We [follow the same conventions for writing JS as used by `govuk-frontend`](https://github.com/alphagov/govuk-frontend/blob/main/docs/contributing/coding-standards/js.md). We lint styles using StandardJS.
+
+Individual JavaScript modules are bundled together and served only to the interface they are used by. This ensures that we only serve JavaScript to the users that require it, keeping the overall size of our assets to a minimum.
+
+| Interface                            | JavaScript entry point                        |
+| ------------------------------------ | --------------------------------------------- |
+| Shared                               | `app/frontend/packs/application.js`           |
+| API docs                             | `app/frontend/packs/application-api-docs.js`  |
+| Apply for teacher training           | `app/frontend/packs/application-candidate.js` |
+| Manage teacher training applications | `app/frontend/packs/application-provider.js`  |
+| Support for apply                    | `app/frontend/packs/application-support.js`   |
+
+### jQuery
+
+Avoid using jQuery. Browser support for ES5 JavaScript is now widespread enough that a library like jQuery is unnecessary.
+
+### Testing
+
+We test JavaScript using Jest. We also have system specs which test JavaScript using a Chromium headless browser and Capybara.
+
+Spec files should be saved alongside the corresponding JavaScript file.
+
+## Asset pipeline
+
+We use the Rails webpack wrapper [Webpacker](https://github.com/rails/webpacker) to compile CSS, images, fonts and JavaScript.
 
 ## Debugging Webpacker
 
-Webpacker sometimes does not give clear indications of what's wrong when it
-does not work.
+Webpacker sometimes does not give clear indications of what’s wrong when it does not work.
 
-**If you see repeated 'Webpacker compiling...' messages in the Rails server
-log**, a good place to start debugging is by running the webpack compiler via
-`bin/webpack`. This will give a much faster feedback loop than making requests
-using a web browser.
+If you see repeated ‘Webpacker compiling…’ messages in the Rails server log, a good place to start debugging is by running the webpack compiler via `bin/webpack`. This will give a much faster feedback loop than making requests using a web browser.
```

Comment:
```suggestion
If you see repeated `Webpacker compiling…` messages in the Rails server log, a good place to start debugging is by running the webpack compiler via `bin/webpack`. This will give a much faster feedback loop than making requests using a web browser.
```